Mathlib/GroupTheory/IndexNormal.lean

```diff
@@ -0,0 +1,71 @@
+/-
+Copyright (c) 2025 Antoine Chambert-Loir. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Antoine Chambert-Loir
+-/
+import Mathlib.Data.Fintype.Perm
+import Mathlib.Data.Nat.Prime.Factorial
+import Mathlib.GroupTheory.Index
+import Mathlib.GroupTheory.Perm.Finite
+
+/-! # Subgroups of small index are normal
+
+* `Subgroup.normal_of_index_eq_smallest_prime_factor`: in a finite group `G`,
+a subgroup of index equal to the smallest prime factor of `Nat.card G` is normal.
+
+* `Subgroup.normal_of_index_two`: in a group `G`, a subgroup of index 2 is normal
+(This does not require `G` to be finite.)
+
+-/
+
+variable {G : Type*} [Group G] {H : Subgroup G} {p : ℕ}
+
+namespace Subgroup
+
+open MulAction MonoidHom Nat
+
+private theorem dvd_prime {r p : ℕ} (hp : p.Prime) (h : r ∣ p !)
+    (hr : ∀ {l : ℕ} (_ : l.Prime) (_ : l ∣ r), p ≤ l) : r ∣ p := by
+  rwa [← Coprime.dvd_mul_right, mul_factorial_pred hp.pos]
+  contrapose! hr
+  obtain ⟨l, hl, hl'⟩ := exists_prime_and_dvd hr
+  rw [dvd_gcd_iff, hl.dvd_factorial, Nat.le_sub_one_iff_lt hp.pos] at hl'
+  exact ⟨l, hl, hl'⟩
```

Comment:
Sounds like a lemma about the factorial worth being public

Reply:
I rewrote it in a more significant way, this is a bit longer to apply but it looks nicer.